## Actividad 1 — Identificación de la fuente

**1. ¿Quién produjo originalmente los datos?**
United Nations.

**2. ¿Qué organización los procesa para su utilización en Our World in Data?**
Our World in Data.

**3. ¿Qué representa la variable que analizarás?**
El cambio en la poblacion.

## Actividad 2 — Carga y exploración de datos

In [1]:
import pandas as pd
import numpy as np

In [2]:
df = pd.read_csv("../data/raw/annual-population-growth/annual-population-growth.csv")

In [3]:
df.head()

,Entity,Code,Year,Annual change in population,Annual population change (Projected)
0,Afghanistan,AFG,1951,103163.0,NaN
1,Afghanistan,AFG,1952,108441.0,NaN
2,Afghanistan,AFG,1953,108919.0,NaN
3,Afghanistan,AFG,1954,111251.0,NaN
4,Afghanistan,AFG,1955,119027.0,NaN


In [4]:
df.shape

(38400, 5)

In [5]:
df.columns.tolist()

['Entity',
 'Code',
 'Year',
 'Annual change in population',
 'Annual population change (Projected)']

In [6]:
df.dtypes

Entity                                      str
Code                                        str
Year                                      int64
Annual change in population             float64
Annual population change (Projected)    float64
dtype: object

In [7]:
df.describe()

,Year,Annual change in population,Annual population change (Projected)
count,38400.000000,1.868800e+04,1.971200e+04
mean,2025.500000,2.063019e+06,8.375481e+05
std,43.300872,9.440859e+06,5.816745e+06
min,1951.000000,-3.315935e+06,-2.291035e+07
25%,1988.000000,1.781000e+03,-9.742750e+03
50%,2025.500000,4.689800e+04,9.350000e+01
75%,2063.000000,3.021128e+05,1.411160e+05
max,2100.000000,9.337137e+07,7.206736e+07


In [8]:
df.isnull().sum()

Entity                                      0
Code                                     1200
Year                                        0
Annual change in population             19712
Annual population change (Projected)    18688
dtype: int64

In [9]:
df["Entity"].nunique()

256

**1. ¿Cuántas filas y columnas tiene el dataset?**
38,400 filas y 5 columnas.

**2. ¿Qué tipo de datos contiene la variable Year?**
Enteros.

**3. ¿Existen valores faltantes?**
Sí, en las columnas Code (1,200), Annual change in population (19,712) y Annual population change Projected (18,688).

**4. ¿Cuántas entidades diferentes contiene el dataset?**
256 entidades.

## Actividad 3 — Estadística descriptiva

Filtro: **Afghanistan**, período **2005–2023**.

In [10]:
df_A = df[(df["Year"] >= 2005) & (df["Year"] <= 2023) & (df["Entity"] == "Afghanistan")]
df_A.head(19)

,Entity,Code,Year,Annual change in population,Annual population change (Projected)
54,Afghanistan,AFG,2005,843919.0,NaN
55,Afghanistan,AFG,2006,1019525.0,NaN
56,Afghanistan,AFG,2007,485752.0,NaN
57,Afghanistan,AFG,2008,572779.0,NaN
58,Afghanistan,AFG,2009,983471.0,NaN
59,Afghanistan,AFG,2010,817986.0,NaN
60,Afghanistan,AFG,2011,1063620.0,NaN
61,Afghanistan,AFG,2012,1212328.0,NaN
62,Afghanistan,AFG,2013,1062673.0,NaN
63,Afghanistan,AFG,2014,1169819.0,NaN


In [11]:
df_A.mean(numeric_only=True)

Year                                      2014.000000
Annual change in population             941795.052632
Annual population change (Projected)              NaN
dtype: float64

In [12]:
df_A.median(numeric_only=True)

Year                                      2014.0
Annual change in population             988328.0
Annual population change (Projected)         NaN
dtype: float64

In [13]:
df_A.var(numeric_only=True)

Year                                    3.166667e+01
Annual change in population             4.446413e+10
Annual population change (Projected)             NaN
dtype: float64

In [14]:
df_A.std(numeric_only=True)

Year                                         5.627314
Annual change in population             210865.183261
Annual population change (Projected)              NaN
dtype: float64

In [15]:
df_A.min(numeric_only=True)

Year                                      2005.0
Annual change in population             485752.0
Annual population change (Projected)         NaN
dtype: float64

In [16]:
df_A.max(numeric_only=True)

Year                                       2023.0
Annual change in population             1212852.0
Annual population change (Projected)          NaN
dtype: float64

**1. ¿Cuál es el valor de la media?**
941,795.05.

**2. ¿Cuál es el valor de la mediana?**
988,328.

**3. ¿Cuál es el valor de la desviación estándar?**
210,865.18.

**4. ¿Qué entidad presenta el valor mínimo y cuál el máximo?**
Mínimo: 485,752 (Afghanistan 2007). Máximo: 1,212,852 (Afghanistan 2020).

## Actividad 4 — Comprobación del entorno

In [17]:
import sys
print("Python:", sys.version)
print("Pandas:", pd.__version__)
print("NumPy: ", np.__version__)

Python: 3.14.3 (tags/v3.14.3:323c59a, Feb  3 2026, 16:04:56) [MSC v.1944 64 bit (AMD64)]
Pandas: 3.0.3
NumPy:  2.4.6
